ospfclient/ospf_apiclient.c

```diff
@@ -217,7 +219,7 @@ struct ospf_apiclient *ospf_apiclient_connect(char *host, int syncport)
 	}
 
 	/* Prepare address structure for connect */
-	memcpy(&myaddr_sync.sin_addr, hp->h_addr, hp->h_length);
+	myaddr_sync = *(struct sockaddr_in *)res->ai_addr;
```

Comment:
The problem is that it's not just `gethostbyname` that's deprecated… it's also "expecting/working with one address". Using `getaddrinfo` like this is in fact worse than `gethostbyname` since it is now random which of (generally multiple because IPv6) addresses you get in the first slot of the result.

If you want to move it forward, you also need to make it support multiple results, otherwise it's just worse than before :disappointed:

Reply:
ipv6 is not being requested.  How will it get more than one answer?